src/cljs/rems/spa.cljs

```diff
@@ -341,61 +333,41 @@
                     (config/fetch-config!))}
     [:i.fas.fa-redo]]])
 
-(defn footer []
+(defn- footer []
   [:footer.footer
    [:div.container.d-flex.flex-row.justify-content-between.align-items-center
     [:div.d-flex.flex-row [nav/footer-extra-pages]]
     [:div.footer-text (text :t/footer)]
     (when (config/dev-environment?)
       [dev-reload-button])]])
 
-(defn main-content [_page-id _grab-focus?]
-  (let [on-update (fn [this]
-                    (let [[_ _page-id grab-focus?] (r/argv this)]
-                      (when grab-focus?
-                        (when-let [element (or (.querySelector js/document "h1")
-                                               (.querySelector js/document "#main-content"))]
-                          (focus/focus element)
-                          (rf/dispatch [::focus-grabbed])))))]
-    (r/create-class
-     {:component-did-mount on-update
-      :component-did-update on-update
-      :display-name "main-content"
-      :reagent-render (fn [page-id _grab-focus?]
-                        [:main.container-fluid
-                         {:class (str "page-" (name page-id))
-                          :id "main-content"}
-                         (if-let [content (pages page-id)]
-                           [:<>
-                            [missing-email-warning]
-                            [content]]
-                           (do ; implementation error
-                             (println "Unknown page-id" page-id)
-                             (rf/dispatch [:set-active-page :not-found])
-                             nil))])})))
+(defn- logo [page-id]
+  (when (or (= :home page-id)
+            (not (rems.theme/use-navbar-logo?)))
+    [:div {:class "logo"}
+     [:div.img]]))
 
-(defn- lazy-load-data!
-  "Loads datasets that are not required for immediate render or e.g. require login."
-  []
-  (when @(rf/subscribe [:user])
-    (when (empty? @(rf/subscribe [:organizations]))
-      (config/fetch-organizations!))))
+(defn- main-content [page-id]
+  (b/cond
+    :when page-id
+    :let [content (get pages page-id)]
+
+    (some? content)
+    [:main#main-content.container-fluid {:class (str "page-" (name page-id))}
+     [missing-email-warning]
+     [content]]
+
+    (do ; implementation error
```

Comment:
Could be nice to keep using `:else` like in regular `cond` as it is a bit more explicit? Not sure.